src/tracks/playabletrack/wavetrack/ui/SpectrumCache.cpp

```diff
@@ -277,7 +268,10 @@ bool SpecCache::CalculateOneSpectrum
                      (numRe * denomRe + numIm * denomIm) / power;
                }
 
-               int correctedX = (floor(0.5 + xx + timeCorrection * pixelsPerSecond / rate));
+               // todo(mhodgkinson) double-check, but I think here we need a raw
+               // sample index - stretchRatio not involved.
```

Comment:
You are right.  `timeCorrection` is scaled to the clip's raw sample rate, without the stretching ratio correction for real time.  We want to find the correct X coordinate for that.

So remove "todo" and explain in the comment more confidently.

Reply:
Reference #2 on this page was my guide to implement this algorithm long ago.  https://en.wikipedia.org/wiki/Reassignment_method#References

I used to be able to find freely available pdfs

Reply:
You have an IEEE subscription ?? :star-struck:

Reply:
Copy-pasting your explanation as comment here, tagging it with your initials.

Reply:
> You have an IEEE subscription ?? :star-struck:

Actually no